proto/internal/temporal/server/api/persistence/v1/executions.proto

```diff
@@ -308,8 +317,23 @@ message ActivityInfo {
     int64 scheduled_event_id = 30;
     temporal.api.common.v1.Payloads last_heartbeat_details = 31;
     google.protobuf.Timestamp last_heartbeat_update_time = 32;
+    // When true, it means the activity is assigned to the build ID of its workflow (only set for old versioning)
+    // Deprecated. use `use_workflow_build_id`
     bool use_compatible_version = 33;
     temporal.api.common.v1.ActivityType activity_type = 34;
+    // Absence of `assigned_build_id` generally means this task is on an "unversioned" task queue.
+    // In rare cases, it can also mean that the task queue is versioned but we failed to write activity's
+    // independently-assigned build ID to the database. This case heals automatically once the task is dispatched.
+    oneof assigned_build_id {
+        // When true, it means this activity is assigned to the build ID of its workflow.
+        google.protobuf.Empty use_workflow_build_id = 35;
+        // This means the activity is independently versioned and not bound to the build ID of its workflow.
+        // If the task fails and is scheduled again, the assigned build ID may change according to the latest versioning
+        // rules.
+        string last_independently_assigned_build_id = 36;
+    }
+    // The version stamp of the worker to whom this activity was most-recently dispatched
+    temporal.api.common.v1.WorkerVersionStamp last_worker_version_stamp = 37;
```

Comment:
nit: shall we use the same name for this and the one in ExecutionInfo L162?

Reply:
I considered that, but in ActivityInfo message we're using "last" in all the other fields so I thought its more important to keep it consistent within the proto message.